hudi-common/src/test/java/org/apache/hudi/common/testutils/HoodieTestDataGenerator.java

```diff
@@ -269,25 +300,25 @@ public static GenericRecord generateGenericRecord(String rowKey, String partitio
     rec.put("partition_path", partitionPath);
     rec.put("rider", riderName);
     rec.put("driver", driverName);
-    rec.put("begin_lat", RAND.nextDouble());
-    rec.put("begin_lon", RAND.nextDouble());
-    rec.put("end_lat", RAND.nextDouble());
-    rec.put("end_lon", RAND.nextDouble());
+    rec.put("begin_lat", r.nextDouble());
+    rec.put("begin_lon", r.nextDouble());
+    rec.put("end_lat", r.nextDouble());
+    rec.put("end_lon", r.nextDouble());
     if (isFlattened) {
-      rec.put("fare", RAND.nextDouble() * 100);
+      rec.put("fare", r.nextDouble() * 100);
       rec.put("currency", "USD");
     } else {
-      rec.put("distance_in_meters", RAND.nextInt());
-      rec.put("seconds_since_epoch", RAND.nextLong());
-      rec.put("weight", RAND.nextFloat());
+      rec.put("distance_in_meters", r.nextInt());
+      rec.put("seconds_since_epoch", r.nextLong());
+      rec.put("weight", r.nextFloat());
       byte[] bytes = "Canada".getBytes();
       rec.put("nation", ByteBuffer.wrap(bytes));
-      long currentTimeMillis = System.currentTimeMillis();
-      Date date = new Date(currentTimeMillis);
+      long randomMillis = genRandomTimeMillis(r);
```

Comment:
One nit for thought.  Previously, each time generateGenericRecord() is called, the timestamp monotonically increases.  If some tests depend on this assumption, the new time millis random gen should also honor that in a way, e.g., start with a random millis, and increment it by another positive random millis for each subsequent call, to guarantee the order.